src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateVirsions = [];
+  const copyState = { ...state };
+
+  for (const action of actions) {
+    Object.assign(copyState, action.extraData);
+
+    switch (action.type) {
+      case 'addProperties':
+        stateVirsions.push(Object.assign({}, copyState));
+        break;
+
+      case 'removeProperties':
+        for (const property of action.keysToRemove) {
```

Comment:
fix your variable name here
```suggestion
        for (const ____ of action.keysToRemove) {
```